MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -0,0 +1,275 @@
+import Polyline
+import MapboxDirections
+import AVFoundation
+import MapboxMobileEvents
+
+let SecondsBeforeCollectionAfterFeedbackEvent: TimeInterval = 20
+
+struct EventDetails {
+    var originalRequestIdentifier: String?
+    var requestIdentifier: String?
+    var coordinate: CLLocationCoordinate2D?
+    var originalGeometry: Polyline?
+    var originalDistance: CLLocationDistance?
+    var originalEstimatedDuration: TimeInterval?
+    var originalStepCount: Int?
+    var geometry: Polyline?
+    var distance: CLLocationDistance?
+    var estimatedDuration: TimeInterval?
+    var stepCount: Int?
+    var created: Date
+    var startTimestamp: String?
+    var platform: String
+    var operatingSystem: String
+    var device: String
+    var sdkIdentifier: String
+    var sdkVersion: String
+    var eventVersion: Int
+    var profile: String
+    var simulation: Bool
+    var sessionIdentifier: String
+    var distanceCompleted: CLLocationDistance
+    var distanceRemaining: TimeInterval
+    var durationRemaining: TimeInterval
+    var rerouteCount: Int
+    var volumeLevel: Int
+    var screenBrightness: Int
+    var batteryPluggedIn: Bool
+    var batteryLevel: Float
+    var applicationState: String
+    
+    init(routeController: RouteController, session: SessionState) {
+        created = Date()
+        if let start = session.departureTimestamp?.ISO8601 {
+            startTimestamp =  start
+        }
+        
+        platform = ProcessInfo.systemName
+        operatingSystem = "\(ProcessInfo.systemName) \(ProcessInfo.systemVersion)"
+        device = UIDevice.current.machine
```

Comment:
`platform`, `operatingSystem`, `sdkVersion`, etc. can never change throughout the session, so set them in `convertedToDictionary()` rather than increasing the size of the struct. The fewer properties in the struct, the smaller each instance of the struct and the higher likelihood that Swift will use an efficient memory representation for it.